contribs/gmf/src/services/objecteditingquery.js

```diff
@@ -0,0 +1,217 @@
+ * @return {Array.<gmf.ObjectEditingQuery.QueryableLayerInfo>} List of
+ *     queryable layers information.
+ * @export
+ */
+gmf.ObjectEditingQuery.getQueryableLayersInfoFromThemes = function(
+  themes, ogcServers
+) {
+  var queryableLayersInfo = [];
+  var theme;
+  var group;
+  var nodes;
+  var node;
+
+  for (var i = 0, ii = themes.length; i < ii; i++) {
+    theme = /** @type {gmfThemes.GmfTheme} */ (themes[i]);
+    for (var j = 0, jj = theme.children.length; j < jj; j++) {
+      group = /** @type {gmfThemes.GmfGroup} */ (theme.children[j]);
+
+      // Skip groups that don't have an ogcServer set
+      if (!group.ogcServer) {
+        continue;
+      }
+
+      nodes = [];
+      gmf.Themes.getFlatNodes(group, nodes);
+
+      for (var k = 0, kk = nodes.length; k < kk; k++) {
+        node = /** @type {gmfThemes.GmfGroup|gmfThemes.GmfLayerWMS} */ (
+          nodes[k]);
+
+        // Skip groups within groups
+        if (node.children && node.children.length) {
+          continue;
+        }
+
+        if (node.childLayers &&
+          node.childLayers[0] &&
+          node.childLayers[0].queryable
+        ) {
+          queryableLayersInfo.push({
+            layerNode: node,
+            ogcServer: ogcServers[group.ogcServer]
+          });
+        }
+      }
+    }
+  }
+
+  return queryableLayersInfo;
+};
+
+
+/**
+ * From a queryable layer (WMS layer node), use its associated OGC server
+ * to issue a single WMS GetFeatureInfo request at a specific location on a
+ * specific map to fetch a single feature. If no feature is found, a `null`
+ * value is returned.
+ *
+ * @param {gmf.ObjectEditingQuery.QueryableLayerInfo} layerInfo Queryable layer
+ *     information.
+ * @param {ol.Coordinate} coordinate Coordinate.
+ * @param {ol.Map} map Map.
+ * @return {angular.$q.Promise} Promise.
+ * @export
+ */
+gmf.ObjectEditingQuery.prototype.getFeatureInfo = function(
+  layerInfo,
+  coordinate,
+  map
+) {
+
+  var view = map.getView();
+  var projCode = view.getProjection().getCode();
+  var resolution = /** @type {number} */(view.getResolution());
+  var infoFormat = ngeo.QueryInfoFormatType.GML;
+  var layerNode = layerInfo.layerNode;
+  var layersParam = layerNode.layers.split(',');
+  var ogcServer = layerInfo.ogcServer;
+
+  var format = new ol.format.WMSGetFeatureInfo({
+    layers: layersParam
+  });
+
+  var wmsSource = new ol.source.ImageWMS({
+    url: ogcServer.url,
+    params: {
+      layers: layersParam
+    }
+  });
+
+  var url = /** @type {string} */ (
+    wmsSource.getGetFeatureInfoUrl(coordinate, resolution, projCode, {
+      'INFO_FORMAT': infoFormat,
+      'FEATURE_COUNT': 1,
+      'QUERY_LAYERS': layersParam
+    })
+  );
+
+  return this.http_.get(url).then(
+    function(format, response) {
+      var features = format.readFeatures(response.data);
+      return (features && features[0]) ? features[0] : null;
+    }.bind(this, format)
+  );
+
```

Comment:
It's the 'heat of the moment' I suppose.  No, seriously I don't really have a rule for this myself and since the linter isn't complaining about it I sometime leave a line and sometime don't.  I don't think we should care about this.  If we do, then we should add it in the rules of the linter to be able.

For now, I'll remove it.